src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -193,5 +193,356 @@ public void ContainsKey(int size)
+
+            for (int i = 0; i < sampleLength; i++)
+            {
+                sampleKeys[i] = i.ToString();
+                sampleValues[i] = i.ToString();
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsKey(sampleKeys[j]);  //Every key searched for is present in the dictionary.
+        }
+
+        [Benchmark]
+        public static void Dictionary_ContainsKey_String_False()
+        {
+            int sampleLength = 10000;
+
+            string[] sampleKeys = new string[sampleLength];
+            string[] sampleValues = new string[sampleLength];
+
+            Dictionary<string, string> dictionary = new Dictionary<string, string>();
+
+            for (int i = 0; i < sampleLength; i++)
+            {
+                sampleKeys[i] = i.ToString();
+                sampleValues[i] = i.ToString();
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            string missingKey = sampleLength.ToString();  //The string representation of sampleLength is not present in the dictionary.
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsKey(missingKey);
+        }
+
+        [Benchmark]
+        public static void Dictionary_ContainsKey_String_False_IgnoreCase()
+        {
+            int sampleLength = 10000;
+
+            string[] sampleKeys = new string[sampleLength];
+            string[] sampleValues = new string[sampleLength];
+
+            Dictionary<string, string> dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+            for (int i = 0; i < sampleLength; i++)
+            {
+                sampleKeys[i] = i.ToString();
+                sampleValues[i] = i.ToString();
+
+                dictionary.Add(sampleKeys[i], sampleValues[i]);
+            }
+
+            bool result = false;
+
+            string missingKey = sampleLength.ToString();  //The string representation of sampleLength is not present in the dictionary.
+
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int j = 0; j < sampleLength; j++)
+                        result = dictionary.ContainsKey(missingKey);
+        }
+
+        private static Dictionary<int?, int?> s_collection;
+        private static int? s_key;
+        private static int? s_value;
+        private static int? s_item;
+
+        private static Dictionary<long?, long?> s_collection1;
```

Comment:
They don't need to be statics as far as i can tell.  FYI, I did not do the original porting, so I may not have all of the context, but I'll be happy to make appropriate changes.

I've fixed this.